### ISSUE TITLE
Owasp Zap sarif plugin shows application/javascript as binary

### ISSUE THREAD

**Issue by de-jcup** [mercedes-benz/zap-extensions#7]
## Situation
When we have `application/javascript` as mime type, we currently have a binary sarif result

## Wanted
shall be source...

## Solution
- Changes source/binary detection parts so application/javascript is accepted as source
- Check for other wrong interpreted mime types 
- Add some tests

**Comment by de-jcup** [comment]
Reading https://www.rfc-editor.org/rfc/rfc4329.txt:

```
   ...
   Various unregistered media types have been used in an ad-hoc fashion
   to label and exchange programs written in ECMAScript and JavaScript.
   These include:

     +-----------------------------------------------------+
      | text/javascript          | text/ecmascript          |
      | text/javascript1.0       | text/javascript1.1       |
      | text/javascript1.2       | text/javascript1.3       |
      | text/javascript1.4       | text/javascript1.5       |
      | text/jscript             | text/livescript          |
      | text/x-javascript        | text/x-ecmascript        |
      | application/x-javascript | application/x-ecmascript |
      | application/javascript   | application/ecmascript   |
      +-----------------------------------------------------+

   Use of the "text" top-level type for this kind of content is known to
   be problematic.  This document thus defines text/javascript and text/
   ecmascript but marks them as "obsolete".  Use of experimental and
   unregistered media types, as listed in part above, is discouraged.
   The media types,

      * application/javascript
      * application/ecmascript

   which are also defined in this document, are intended for common use
   and should be used instead.
```

But here we should also provide the "obsolete" parts to have even unofficial MIME-types supported.

**Comment by de-jcup** [comment]
closed with 6afe1b0 in branch `sarif-reporting`
